File: 977. Squares of a Sorted Array/977.js
// 977. Squares of a Sorted Array solution in JavaScript
// https://leetcode.com/problems/squares-of-a-sorted-array/description/

/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function(nums) {
    let left = 0;
    let right = nums.length - 1;
    let result = new Array(nums.length);

    for (let i = nums.length - 1; i >= 0; i--) {
        let square = 0;

        if (Math.abs(nums[left]) > Math.abs(nums[right])) {
            square = nums[left] * nums[left];
            left++;
        } else {
            square = nums[right] * nums[right];
            right--;
        }

        result[i] = square;
    }

    return result;
};

File: 977. Squares of a Sorted Array/977.java
// 977. Squares of a Sorted Array solution in Java
// https://leetcode.com/problems/squares-of-a-sorted-array/description/

class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        
        int[] final_array = new int[n];
        
        
        
        int left = 0;
        int right = nums.length - 1;
        
        for (int i = n - 1; i >= 0; i--) {
            int square;
            
            if (Math.abs(nums[right]) > Math.abs(nums[left])) {
                square = nums[right] * nums[right];
                
                right--;
            } else {
                square = nums[left] * nums[left];
                left++;
            }
            
            final_array[i] = square;
        }
        
        return final_array;
    }
}